src/shogun/kernel/GaussianKernel.cpp

```diff
@@ -121,3 +121,18 @@ float64_t GaussianKernel::distance(int32_t idx_a, int32_t idx_b) const
 {
 	return ShiftInvariantKernel::distance(idx_a, idx_b)/get_width();
 }
+
+SGMatrix<float64_t> GaussianKernel::sample_spectral_density(int32_t dim_output, int32_t dim_input_space) const
+{
+	NormalDistribution<float64_t> normal_dist;
+	SGMatrix<float64_t> sampled_kernel(dim_output, dim_input_space);
+	const auto width = get_width();
+	const auto std_dev = std::sqrt(2.0 / width);
```

Comment:
imo this would be `1 / width` as the gaussian kernel in shogun is exp(-sq_dists / width)

Also see https://github.com/karlnapf/kernel_exp_family/blob/master/kernel_exp_family/kernels/kernels.py

Reply:
(which kind of makes me worried that this is wrong in the last PR as well)

Reply:
Please double check the tests where you compare shogun's gk output the to rff approximate kernel output...I am pretty sure they are different

Reply:
Check sklearn's implementation in this [link](https://github.com/scikit-learn/scikit-learn/blob/fd237278e895b42abe8d8d09105cbb82dc2cbba7/sklearn/kernel_approximation.py#L111)

Their ```gamma = 1/width``` , 
and they've used
 ```np.sqrt(2.* gamma)``` 
we've used
 ```std::sqrt(2.0 / width)```

I've checked-rechecked our code more times than I can count :).

Reply:
Looks like your are right. I really wonder what my other code has been doing then all these years :D

We should double check though, that the error of kernel - rff_kernel goes to zero after this refactor !